src/modules/update/update-common/http.c

```diff
@@ -109,6 +109,8 @@ task_get_metadata_response(void *data, const struct sol_http_client_connection *
     handle->conn = NULL;
     handle->on_callback = true;
 
+    SOL_HTTP_RESPONSE_CHECK_API(http_response);
+
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file